api/grpcserver/v2alpha1/transaction_test.go

```diff
@@ -474,3 +479,224 @@ func newTx(nonce uint64, recipient types.Address, signer *signing.EdSigner) *typ
+		require.Nil(t, contents.GetSingleSigSpawn())
+		require.Nil(t, contents.GetVestingSpawn())
+		require.Nil(t, contents.GetVaultSpawn())
+		require.Nil(t, contents.GetDrainVault())
+		require.Equal(t, spacemeshv2alpha1.Transaction_TRANSACTION_TYPE_MULTI_SIG_SEND, txType)
+	})
+
+	t.Run("vault spawn", func(t *testing.T) {
+		t.Parallel()
+
+		var pubs []ed25519.PublicKey
+		pks := make([]ed25519.PrivateKey, 0, 3)
+		for i := 0; i < 3; i++ {
+			pub, pk, err := ed25519.GenerateKey(rand.New(rand.NewSource(time.Now().UnixNano())))
+			require.NoError(t, err)
+			pubs = append(pubs, pub)
+			pks = append(pks, pk)
+		}
+
+		owner := wallet.Address(pubs[0])
+		vaultArgs := &vault.SpawnArguments{
+			Owner:               owner,
+			InitialUnlockAmount: uint64(1000),
+			TotalAmount:         uint64(1001),
+			VestingStart:        105120,
+			VestingEnd:          4 * 105120,
+		}
+		vaultAddr := core.ComputePrincipal(vault.TemplateAddress, vaultArgs)
+
+		var agg *multisig2.Aggregator
+		for i := 0; i < len(pks); i++ {
+			part := multisig2.Spawn(uint8(i), pks[i], vaultAddr, vault.TemplateAddress, vaultArgs, types.Nonce(0))
+			if agg == nil {
+				agg = part
+			} else {
+				agg.Add(*part.Part(uint8(i)))
+			}
+		}
+		rawTx := agg.Raw()
+
+		contents, txType, err := toTxContents(rawTx)
+		require.NoError(t, err)
+		require.NotNil(t, contents.GetVaultSpawn())
+		require.Nil(t, contents.GetMultiSigSpawn())
+		require.Nil(t, contents.GetSingleSigSpawn())
+		require.Nil(t, contents.GetVestingSpawn())
+		require.Nil(t, contents.GetSend())
+		require.Nil(t, contents.GetDrainVault())
+		require.Equal(t, spacemeshv2alpha1.Transaction_TRANSACTION_TYPE_VAULT_SPAWN, txType)
+	})
+
+	t.Run("drain vault", func(t *testing.T) {
+		t.Parallel()
+
+		var pubs [][]byte
+		pks := make([]ed25519.PrivateKey, 0, 3)
+		for i := 0; i < 3; i++ {
+			pub, pk, err := ed25519.GenerateKey(rand.New(rand.NewSource(time.Now().UnixNano())))
+			require.NoError(t, err)
+			pubs = append(pubs, pub)
+			pks = append(pks, pk)
+		}
+
+		principal := multisig2.Address(multisig.TemplateAddress, 3, pubs...)
+		to := wallet.Address(pubs[1])
+		vaultAddr := wallet.Address(pubs[2])
+
+		agg := vesting.DrainVault(
+			0,
+			pks[0],
+			principal,
+			vaultAddr,
+			to,
+			100,
+			types.Nonce(1))
+		for i := 1; i < len(pks); i++ {
+			part := vesting.DrainVault(uint8(i), pks[i], principal, vaultAddr, to, 100, types.Nonce(1))
+			agg.Add(*part.Part(uint8(i)))
+		}
+		rawTx := agg.Raw()
+
+		contents, txType, err := toTxContents(rawTx)
+		require.NoError(t, err)
+		require.NotNil(t, contents.GetDrainVault())
+		require.Nil(t, contents.GetMultiSigSpawn())
+		require.Nil(t, contents.GetSingleSigSpawn())
+		require.Nil(t, contents.GetVestingSpawn())
+		require.Nil(t, contents.GetSend())
+		require.Nil(t, contents.GetVaultSpawn())
+		require.Equal(t, spacemeshv2alpha1.Transaction_TRANSACTION_TYPE_DRAIN_VAULT, txType)
+	})
+
+	t.Run("multisig spawn", func(t *testing.T) {
```

Comment:
There is already a test with the same name.
```suggestion
	t.Run("multisig vesting spawn", func(t *testing.T) {
```